client/landing/jetpack-cloud/components/filter-bar/index.tsx

```diff
@@ -0,0 +1,171 @@
+	);
+
+	const toggleIsActivityTypeSelectorVisible = () => {
+		setIsDateRangeSelectorVisible( false );
+		setIsActivityTypeSelectorVisible( ! isActivityTypeSelectorVisible );
+	};
+
+	const closeActivityTypeSelector = () => {
+		setIsActivityTypeSelectorVisible( false );
+	};
+
+	const toggleIsDateRangeSelectorVisible = () => {
+		setIsActivityTypeSelectorVisible( false );
+		setIsDateRangeSelectorVisible( ! isDateRangeSelectorVisible );
+	};
+
+	const closeDateRangeSelector = () => {
+		setIsDateRangeSelectorVisible( false );
+	};
+
+	const activityTypeButtonRef = useRef< Button >( null );
+	const dateRangeButtonRef = useRef< Button >( null );
+
+	// const closeDateRangeSelector = () => {
+	// 	setIsDateRangeSelectorVisible( false );
+	// };
+
+	const getButtonClassName = ( isActive: boolean, hasSelections: boolean ) => {
+		if ( hasSelections ) {
+			return 'filter-bar__button-with-selections';
+		}
+
+		return isActive ? 'filter-bar__button-active' : 'filter-bar__button';
+	};
+
+	const renderActivityTypeSelectorButtonText = ( groups?: string[] ) =>
+		groups
+			? translate( 'Activity type: %d selected', {
+					args: [ groups.length ],
+			  } )
+			: translate( 'Activity type' );
+
+	const onGroupsChange = ( newGroups: string[] ) =>
+		onFilterChange( { ...filter, group: newGroups.length > 0 ? newGroups : undefined } );
+
+	// when the filter changes re-request the activity counts
+	// the activity counts only use the date values, but the underlying data layer handles unnecessary re-requests via freshness
+	useEffect( () => {
+		requestActivityActionTypeCounts( siteId, filter );
+	}, [ filter, siteId ] );
+
+	const render = () => (
+		<>
+			<p>{ translate( 'Filter by:' ) }</p>
+			{ showActivityTypeSelector && (
+				<>
+					<div className="filter-bar__button-group">
+						<Button
+							className={ getButtonClassName(
+								isActivityTypeSelectorVisible,
+								filter.group !== undefined
+							) }
+							compact
+							onClick={ toggleIsActivityTypeSelectorVisible }
+							ref={ activityTypeButtonRef }
+						>
+							{ renderActivityTypeSelectorButtonText( filter.group ) }
+						</Button>
+						{ filter.group && (
+							<Button
+								className="filter-bar__clear-selection-button"
+								compact
+								onClick={ () => onGroupsChange( [] ) }
+							>
+								<Gridicon icon="cross" />
+							</Button>
+						) }
+					</div>
+					<ActivityTypeSelector
+						activityCounts={ activityActionTypeCounts }
+						context={ activityTypeButtonRef }
+						groups={ filter.group || [] }
+						isVisible={ isActivityTypeSelectorVisible }
+						onClose={ closeActivityTypeSelector }
+						onGroupsChange={ onGroupsChange }
+					/>
+				</>
+			) }
+			{ showDateRangeSelector && (
+				<>
+					<Button
+						className={ getButtonClassName( isDateRangeSelectorVisible, false ) }
+						compact
+						onClick={ toggleIsDateRangeSelectorVisible }
+						ref={ dateRangeButtonRef }
+					>
+						{ translate( 'Date range' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **127** times:
`translate( 'Date range:' )` *ES Score: 10*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-cloud-controlled-filterbar)